### DISCUSSION TITLE
Typo in `pyright: ignore` regex?

### DISCUSSION THREAD

**Discussion by InSyncWithFoo** [microsoft/pyright#8000] [General]
Currently, `pyright: ignore` comments are matched using [the following regular expression](https://github.com/microsoft/pyright/blob/c1b5bccca05542d1119a37fe3c54e0588cf0f0dc/packages/pyright-internal/src/parser/tokenizer.ts#L1302):

```js
/((^|#)\s*)pyright:\s*ignore(\s*\[([\s*\w-,]*)\]|\s|$)/
```

That `\s*` in `[\s*\w-,]*` looks very much like an error. If it isn't one, that suggests the rule list may contain literal `*` (e.g. `pyright: ignore[*foo*, b*r]`). However, there are no such rules as far as I'm aware.

Is this intentional?

**Comment by erictraut** [comment]
Yes, this is a bug. Thanks for reporting.

I've fixed it for the `# pyright: ignore` and `# type: ignore` cases.
